### ISSUE TITLE
twig: Show in the template selection where the selection is taken from

### ISSUE THREAD

**Issue by mpitz** [contao/contao#6481]
### Affected version(s)

4.13.31

### Description

POV: Creating a custom CTE

Default Template: `contao/templates/content_element/my_template.html.twig`
Theme Template: `templates/my_theme/content_element/my_template.html.twig`

If you use the .html5 (PHP) templates, the template selection shows that the default template is used. If the template is then also created in the theme folder, the following is displayed: (Standard, [THEME NAME]).

With the twig templates, only [App] is displayed. However, the template that is in the theme folder is obviously used.

Wouldn't it make sense to show exactly which template is meant in the display for the twig templates as well?

**Comment by fritzmg** [comment]
> With the twig templates, only [App] is displayed. However, the template that is in the theme folder is obviously used.

You mean, you only have the following files for example:

```
contao/templates/.twig-root
contao/templates/content_element/foobar.html.twig
templates/theme/content_element/foobar.html.twig
```

?

**Comment by mpitz** [comment]
Yes, Sir!

**Comment by m-vo** [comment]
Whether a theme template is used, is completely a runtime decision and not something you can actively select. That's also why you always need a "non-theme" version of a variant template for it to show.

**Comment by mpitz** [comment]
Yes, I understand that. This is not a direct variant, but a theme template that overwrites the default template. And actually it is only about the display in the template selection that does not show that the theme template is used.

**Comment by fritzmg** [comment]
> Yes, Sir!

You will need to add

```
templates/content_element/foobar.html.twig
```

**Comment by mpitz** [comment]
> > Yes, Sir!
> 
> You will need to add
> 
> ```
> templates/content_element/foobar.html.twig
> ```

But that would no longer have any relation to the theme.

**Comment by m-vo** [comment]
> And actually it is only about the display in the template selection that does not show that the theme template is used.

Yes, because whether the theme template is used depends on the page/layout context during rendering. The filesystem loader just gives you another result in this case. But you still have the same name in the render call (i.e. it's completely transparent). 

From the perspective of the (static) template hierarchy, theme templates do not exist. If you could select something and then would change the layout/theme or you would reuse the element in a different context, things would be broken / templates would not exist.

**Comment by fritzmg** [comment]
> But that would no longer have any relation to the theme.

That does not matter. iirc you will need to always add the base template override, otherwise the label will not show in the back end.

**Comment by mpitz** [comment]
So you can't see in the backend whether the template was taken from the theme or from the default. Because in both cases the same is displayed in the template selection.

**Comment by fritzmg** [comment]
> Whether a theme template is used, is completely a runtime decision and not something you can actively select. That's also why you always need a "non-theme" version of a variant template for it to show.

It's not about the selection, it's just about the label itself. With PHP templates you get the info that a template override (or variant) is present for a specific theme. Showing this information should still be possible, as it does not depend on any context.
